pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -331,6 +343,53 @@ func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node) error
 	return nil
 }
 
+func createPeerDNSSubjectAlternativeNamesFromMemberList(etcdDiscoveryDomain string, etcdMembers []*etcdserverpb.Member, nodeInternalIPs []string) []string {
+	var found bool
+	var errs []string
+	hosts := []string{"localhost"}
+	for _, m := range etcdMembers {
+		// 4.3 naming
+		if strings.HasPrefix(m.Name, "etcd-member") {
+			found = true
+		}
+		if len(m.PeerURLs) > 0 && len(m.ClientURLs) > 0 {
+			for _, p := range m.PeerURLs {
+				peer, err := url.Parse(p)
+				if err != nil {
```

Comment:
If we are adding the clinet hostname from the clientURLs, do we have to check this peerURLs too?

Reply:
Not sure I am following the question. Here we parse the url for peers. If this is an IP we continue. Basically, if the PeerURL is already IP then we don't need to do any more work.

Reply:
OK. Got it.

Reply:
OK. Thanks. Your test examples make it clear.